app/services/ucas_matches/send_ucas_match_initial_emails_duplicate_applications.rb

```diff
@@ -10,12 +10,18 @@ def call
       raise "Initial emails for UCAS match ##{ucas_match.id} were already sent" if ucas_match.initial_emails_sent?
 
       application_choice = ucas_match.application_choices_for_same_course_on_both_services.first
+      report_bad_data!(ucas_match) if application_choice.blank?
 
       CandidateMailer.ucas_match_initial_email_duplicate_applications(application_choice).deliver_later
 
       application_choice.provider.provider_users.each do |provider_user|
         ProviderMailer.ucas_match_initial_email_duplicate_applications(provider_user, application_choice).deliver_later
       end
     end
+
+    def report_bad_data!(ucas_match)
+      ucas_match.update!(action_taken: :no_application_choice)
+      raise "No application choices found for UCAS match ##{ucas_match.id}"
```

Comment:
What will this mean for the sidekiq failures, will we still see a failure for the job which will then retry? However we won't see those again on a subsequent import as they will pass the resolved stage. 

What is the plan to revisit those matches once the UCAS issue is resolved, will we need to reset those values?

Reply:
The idea is we still get a Sentry notification but we automatically resolve such cases, because there's nothing we can do with them. For the pending retries on QA, this means we'll still get one last notification for each pending message, before they are automatically resolved. The Retries queue will be cleared when this is deployed.